test/bats/azure.bats

```diff
@@ -175,6 +175,11 @@ setup() {
   result_base64_encoded=$(echo "${result//$'\r'}" | base64 -d | openssl pkcs12 -nodes -passin pass:"" | sed -ne '/-BEGIN PRIVATE KEY-/,/-END PRIVATE KEY-/p; /-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' | base64 ${BASE64_FLAGS})
   diff  <(echo "${result_base64_encoded}" ) <(echo "${CERT2_SECRET_VALUE}")
   [[ "${result_base64_encoded}" == "${CERT2_SECRET_VALUE}" ]]
+
+  result=$(kubectl exec nginx-secrets-store-inline-crd-certs -- $EXEC_COMMAND/$CERT2_NAME-secret-pfx-binary | base64)
```

Comment:
nit: 

We can skip the `base64 -d` in `result_base64_encoded` instead of encoding it and then decoding.  

```bash
  result=$(kubectl exec nginx-secrets-store-inline-crd-certs -- $EXEC_COMMAND/$CERT2_NAME-secret-pfx-binary)
  result_base64_encoded=$(echo "${result//$'\r'}" | openssl pkcs12 -nodes -passin pass:"" | sed -ne '/-BEGIN PRIVATE KEY-/,/-END PRIVATE KEY-/p; /-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' | base64 ${BASE64_FLAGS})
  [[ "${result_base64_encoded}" == "${CERT2_SECRET_VALUE}" ]]

```

Reply:
I did try that first, but running just 'cat <binary file>' in kubectl exec corrupted the binary data, and the next call to openssl failed with: asn1 encoding routines:asn1_d2i_read_bio:not enough data

Encoding the file inside the exec statement was enough to get around this so I went with that.  If we want to do this without the base64 encoding I think we'd want to use 'kubectl cp' instead of 'exec'.  That adds a bit of complexity around local file handling in the azure.bats file, but it's something I could look into

Reply:
Thank you for the explanation. That makes sense. We can keep it the way it is now.